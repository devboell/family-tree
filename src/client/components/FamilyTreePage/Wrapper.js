import styled from 'styled-components'
import { pageBlock } from 'globalStyles'


const Wrapper = styled.div`
  ${pageBlock}
`

export default Wrapper
