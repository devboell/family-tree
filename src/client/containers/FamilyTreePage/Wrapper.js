import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
`

export default Wrapper
