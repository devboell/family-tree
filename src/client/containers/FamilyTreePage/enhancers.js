import { gql, graphql, compose } from 'react-apollo'

// TODO: unused for now
const personsQuery = gql`
  query personsQuery {
    persons {
      id
      name
    }
  }
`
const familyTreeQuery = gql`
  query familyTreeQuery {
    familyTree (id: 1)
  }
`
export default compose(
  graphql(personsQuery, {
    name: 'personsQuery',
    options: {
      fetchPolicy: 'network-only',
    },
  }),
  graphql(familyTreeQuery, {
    name: 'familyTreeQuery',
    options: {
      fetchPolicy: 'network-only',
    },
  }),
)
