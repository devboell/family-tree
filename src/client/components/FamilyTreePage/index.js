import React from 'react'

import PersonList from 'containers/PersonList'
import FamilyTree from 'containers/FamilyTree'

import Wrapper from './Wrapper'

const FamilyTreePage = () =>
  <Wrapper>
    <PersonList>
      <FamilyTree />
    </PersonList>
  </Wrapper>

export default FamilyTreePage
