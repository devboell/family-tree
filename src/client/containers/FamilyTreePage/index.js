import React from 'react'
import { PropTypes } from 'prop-types'
import SVGTree from 'components/SVGTree'
import { transformTree } from 'lib/TransformTree'
import enhancers from './enhancers'
import Wrapper from './Wrapper'

const dimensions = {
  boxWidth: 80,
  boxHeight: 60,
  marginX: 30,
  marginY: 50,
}

const FamilyTreePage = ({
  familyTreeQuery,
  familyTreeQuery: familyTree,
}) => (
  familyTreeQuery.loading
  ? <div>loading ...</div>
  :
  <Wrapper>
    <SVGTree
      treeData={transformTree(familyTree.familyTree, dimensions)}
      {...{ dimensions }}
    />
  </Wrapper>
)

FamilyTreePage.propTypes = {
  familyTreeQuery: PropTypes.shape({}).isRequired,
}

export default enhancers(FamilyTreePage)
