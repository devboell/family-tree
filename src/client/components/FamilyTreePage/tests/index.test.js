import React from 'react'
import FamilyTreePage from '../index'

it('FamilyTreePage component, it renders without crashing', () => {
  const wrapper = shallow(<FamilyTreePage />)
  expect(wrapper).toMatchSnapshot()
})
