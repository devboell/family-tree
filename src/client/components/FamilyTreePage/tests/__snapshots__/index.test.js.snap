// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilyTreePage component, it renders without crashing 1`] = `
<styled.div>
  <Connect(Apollo(PersonList))>
    <Apollo(Component) />
  </Connect(Apollo(PersonList))>
</styled.div>
`;
